File: fe/src/types/match.ts
export interface MatchDto {
  id: number;
  homeTeam: {
    id: number;
    name: string;
    code: string;
  };
  awayTeam: {
    id: number;
    name: string;
    code: string;
  };
  stadium: {
    id: number;
    name: string;
  };
  matchDate: string;
  homeTeamScore: number | null;
  awayTeamScore: number | null;
  status: 'SCHEDULED' | 'IN_PROGRESS' | 'FINISHED' | 'CANCELLED';
}

export interface MatchEvent {
  id: number;
  matchId: number;
  type: 'goal' | 'yellow_card' | 'red_card' | 'substitution' | 'var';
  player: string;
  team: 'home' | 'away';
  minute: number;
  description: string;
}

File: be/src/main/kotlin/io/be/entity/Match.kt
package io.be.entity

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "matches")
data class Match(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "home_team_id", nullable = false)
    val homeTeam: Team,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "away_team_id", nullable = false)
    val awayTeam: Team,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "stadium_id", nullable = false)
    val stadium: Stadium,

    @Column(name = "match_date", nullable = false)
    val matchDate: LocalDateTime,

    @Column(name = "home_team_score")
    val homeTeamScore: Int? = null,

    @Column(name = "away_team_score")
    val awayTeamScore: Int? = null,

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    val status: MatchStatus = MatchStatus.SCHEDULED,

    @Column(name = "created_at", updatable = false)
    val createdAt: LocalDateTime = LocalDateTime.now(),

    @Column(name = "updated_at")
    val updatedAt: LocalDateTime = LocalDateTime.now()
)

enum class MatchStatus {
    SCHEDULED, IN_PROGRESS, COMPLETED, CANCELLED
}
